### PR TITLE
MES-998 Locking down WR screen to passcode/fingerprint

### PR DESCRIPTION
**Pull request by roberthall12**
## Description and relevant Jira numbers

Added lock screen trigger to the waiting room page when trying to navigate away from it.
https://jira.i-env.net/browse/MES-998

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

**Comment by Jason-Pullen** [comment]
don't forget the checklists, reminds you to do certain things.

File: src/providers/lock-screen/__tests__/lock-screen.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { LockScreenProviderMock } from '../__mocks__/lock-screen.mock';
+
+describe('Lock Screen Provider', () => {
+
+  let lockScreenProvider: LockScreenProvider;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+                { provide: LockScreenProvider, useClass: LockScreenProviderMock },
```
Comment (johnmorleybjss) [inline src/providers/lock-screen/__tests__/lock-screen.spec.ts]: Spacing is a little off here.

File: src/providers/lock-screen/lock-screen.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+declare let cordova: any;
+
+@Injectable()
+export class LockScreenProvider {
+
+  constructor(
+    private platform: Platform,
+  ) { }
+
+  triggerLockScreen = async (): Promise<any> => {
```
Comment (johnmorleybjss) [inline src/providers/lock-screen/lock-screen.ts]: Could make the type here `Promise<boolean>`.

File: src/providers/lock-screen/__tests__/lock-screen.spec.ts (hunk 1 above)
Comment (ammarhaiderbjss) [inline src/providers/lock-screen/__tests__/lock-screen.spec.ts]: Not much point in the test if you're mocking the thing your testing. You'll need to mock out the cordova plugin instead and use the real `LockScreenProvider`. 
Have a look at the `DeviceProvider` and its test for an example. You'll also need to create a mock for the cordova object in `karmaGlobalMocks.js`.

File: src/providers/lock-screen/lock-screen.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { Platform } from 'ionic-angular';
+
+declare let cordova: any;
+
+@Injectable()
+export class LockScreenProvider {
```
Comment (ammarhaiderbjss) [inline src/providers/lock-screen/lock-screen.ts]: "Lock screen" isn't the clearest name for this I think. We have numerous types of screen "locking", e.g. orientation, prevent sleep, guided access etc. Could we change it to mirror what the cordova plugin was called? So `DeviceAuthenticationProvider`?

File: src/app/types/device-authentication/index.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ * ...
+ *
+ *
+ * this.deviceAuthentication.runAuthentication('Subtitle')
+ *     .then((isAuthenticated)=> console.log(isAuthenticated))
+ *     .catch((errorMsg) => console.log('error: ' + errorMsg))
+ *
+ * ```
+ */
+@Plugin({
```
Comment (ammarhaiderbjss) [inline src/app/types/device-authentication/index.ts]: Not sure what this is for? Can't see it used anywhere?

**Pull request merged**